app/controllers/admin_controller.rb

```diff
@@ -1,5 +1,18 @@
 class AdminController < ApplicationController
+  before_action :set_user
+  before_action :authenticate_admin!
+
   def index
     @total_orders = Order.count
   end
+
+  private
+
+  def authenticate_admin!
+    redirect_to store_index_path, notice: "You don't have privilege to access this section" unless @user.role == "admin"
```

Comment:
translation
enum methods

Reply:
Will do the translation in the next extension 'Mailers and I18n'